pyroute2/netlink/__init__.py

```diff
@@ -1678,6 +1678,79 @@ class mpls_target(target):
 
         family = AF_MPLS
 
+    class seg6_target(target):
+
+        __slots__ = ()
+
+        SEG6_IPTUN_MODE_INLINE = 0
+        SEG6_IPTUN_MODE_ENCAP = 1
+
+        encapmodes = {
+          "inline": SEG6_IPTUN_MODE_INLINE,
+          "encap": SEG6_IPTUN_MODE_ENCAP
+        }
+
+        SEG6_TYPE = 4
+
+        def encode(self):
+
+            # Retrieve the family
+            family = self.get_family()
+            # Seg6 can be applied only to IPv6
+            if family == AF_INET6:
+
+                # TODO other seg6 functionalities:
+                # Inline mode
+                # flags
+                # hmac
+                # ...
+
+                # Get the encap
+                encap = self.value[0]
+                # Get mode
+                mode = encap.get('mode', None)
+                # Get segs
+                segs = encap.get('segs', [])
+                # Initialize value
+                self['value'] = b''
+                # Add mode to value
+                self['value'] += struct.pack('i', self.encapmodes.get(mode, 1))
+                # Nexthdr is not set
+                nexthdr = 0
```

Comment:
is it always == 0?

Reply:
Long story short, they used the SR header (Segment Routing) of the IP packets to instruct the kernel about the tunnel. Nexthdr makes sense only for IP packets where after the IP header there is always something else. Instead for netlink it is not used.